code/modules/wiremod/components/atom/remotecam.dm

```diff
@@ -0,0 +1,454 @@
+	if(!parent || !parent.display_name || parent.display_name == "")
+		shell_camera.c_tag = "[shell_name]: unspecified #[c_tag_random]"
+		current_camera_name = ""
+	else if(current_camera_name != parent.display_name)
+		current_camera_name = parent.display_name
+		var/new_cam_name = reject_bad_name(current_camera_name, allow_numbers = TRUE, ascii_only = FALSE, strict = TRUE, cap_after_symbols = FALSE)
+		//Set camera name using parent circuit name
+		if(new_cam_name)
+			shell_camera.c_tag = "[shell_name]: [new_cam_name] #[c_tag_random]"
+		else
+			shell_camera.c_tag = "[shell_name]: unspecified #[c_tag_random]"
+
+	if(!network.value || network.value == "")
+		shell_camera.network = list("ss13")
+		current_camera_network = ""
+	else if(current_camera_network != network.value)
+		current_camera_network = network.value
+		var/new_net_name = LOWER_TEXT(sanitize(current_camera_network))
+		//Set camera network string
+		if(new_net_name)
+			shell_camera.network = list("[new_net_name]")
+		else
+			shell_camera.network = list("ss13")
+
+/**
+ * Update the chunk for the camera (if enabled)
+ */
+/obj/item/circuit_component/compare/remotecam/proc/update_camera_location(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	SIGNAL_HANDLER
+	if(current_camera_state)
+		GLOB.cameranet.updatePortableCamera(shell_camera, 0.5 SECONDS)
+
+/**
+ * Set the camera as emp'd
+ */
+/obj/item/circuit_component/compare/remotecam/proc/set_camera_emp(datum/source, severity, protection)
+	if(current_camera_emp)
+		return
+	if(!prob(150 / severity))
+		return
+	current_camera_emp = TRUE
+	close_camera()
+	current_camera_emp_timer_id = addtimer(CALLBACK(shell_camera, PROC_REF(post_emp_reset)), REMOTECAM_EMP_RESET, TIMER_STOPPABLE)
+	for(var/mob/M as anything in GLOB.player_list)
+		if (M.client?.eye == shell_camera)
+			M.reset_perspective(null)
+			to_chat(M, span_warning("The screen bursts into static!"))
+
+/**
+ * Restore emp'd camera
+ */
+/obj/item/circuit_component/compare/remotecam/proc/post_emp_reset()
+	current_camera_emp = FALSE
+
+/**
+ * Adds the component to the SSclock_component process list
+ *
+ * Starts draining cell per second while camera is active
+ */
+/obj/item/circuit_component/compare/remotecam/proc/start_process()
+	START_PROCESSING(SSclock_component, src)
+
+/**
+ * Removes the component to the SSclock_component process list
+ *
+ * Stops draining cell per second
+ */
+/obj/item/circuit_component/compare/remotecam/proc/stop_process()
+	STOP_PROCESSING(SSclock_component, src)
+
+/obj/item/circuit_component/compare/remotecam/bci
+	display_name = "BCI Camera"
+	desc = "Digitizes user's sight for surveillance-on-the-go. User must have fully functional eyes for digitizer to work. Camera range input is either 0 (near) or 1 (far). Network field is used for camera network."
+	category = "BCI"
+
+	required_shells = list(/obj/item/organ/internal/cyberimp/bci)
+
+	var/obj/item/organ/internal/cyberimp/bci/bci = null
```

Comment:
There isn't really any particular need for this to be set, you aren't using any of the specific variables on the bci type, so just use the `shell` variable instead.